account_payment_term_partner_holiday/models/res_partner.py

```diff
@@ -0,0 +1,87 @@
+# Copyright 2021 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import _, api, fields, models
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    holiday_ids = fields.One2many(
+        comodel_name="res.partner.holiday",
+        inverse_name="partner_id",
+        string="Holidays",
+        copy=True,
+        auto_join=True
+    )
+
+    def is_date_in_holiday(self, date):
+        if isinstance(date, str):
+            date = fields.Date.from_string(date)
+        res = self.holiday_ids.filtered(
+            lambda x: int(x.month_from) <= date.month and int(x.month_to) >= date.month
+            and int(x.day_from) <= date.day and int(x.day_to) >= date.day
+        )
+        if bool(res):
+            res = res[0]
+            return [
+                fields.Date.from_string(
```

Comment:
One of this two can give an error when doing the conversion, so you should sanitize returning the minimum allowed date. Example: you put 31 of February, you should return 28 or 29 depending on the year. It can be an iterating method that substract one while `from_string` gives error.

Reply:
Ok, i have change it and add February in test to test it.